src/App.tsx

```diff
@@ -14,11 +14,116 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
-
-export default App;
+interface State {
+  selectedGoodsList: string[],
+}
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGoodsList: ['Jam'],
+  };
+
+  selectGoods = (item: string) => {
+    this.setState((state: State) => {
+      state.selectedGoodsList.push(item);
+
+      return { selectedGoodsList: state.selectedGoodsList };
+    });
+  };
+
+  removeGood = (item: string) => {
+    this.setState((state: State) => {
+      state.selectedGoodsList.splice(state.selectedGoodsList.indexOf(item), 1);
+
+      return { selectedGoodsList: state.selectedGoodsList };
+    });
+  };
+
+  getGoodsString = (): string => {
+    const { selectedGoodsList } = this.state;
+    let resultString = '';
+
+    switch (true) {
+      case (selectedGoodsList.length === 0):
+        resultString = 'No goods selected. Choose one or more:';
+        break;
+      case (selectedGoodsList.length === 1):
+        resultString = `${selectedGoodsList[0]} is selected`;
+        break;
+      case (selectedGoodsList.length === 2):
+        resultString = `${selectedGoodsList.join(' and ')} are selected`;
+        break;
+      case (selectedGoodsList.length === 10):
```

Comment:
Nice, but what if tomorrow we add one more good?

Reply:
thanks) it wiil be
selectedGoodsList.length === goodsFromServer.length 😄